### PR TITLE
Add experimental resources

### PR DESCRIPTION
**Pull request by stikkireddy**
```hcl
resource "databricks_storage_credential" "this" {
  name = "abc"
  aws_iam_role {
    role_arn = "arn:aws:iam::########:role/some-role"
  }
  comment = "Managed by TF"
}
```

### CODE REVIEW

File: catalog/resource_storage_credentials.go [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+	MetastoreID string                 `json:"metastore_id,omitempty" tf:"computed"`
+	CreatedBy   string                 `json:"created_by,omitempty" tf:"computed"`
+}
+
+//func (a StorageCredentialsAPI) listStorageCredentials() (scis []StorageCredentialConfig, err error) {
+//	err = a.client.Get(a.context, "/unity-catalog/storage-credentials", nil, &scis)
+//	return
+//}
+
+func (a StorageCredentialsAPI) create(cm StorageCredentialConfig) (mi StorageCredentialConfig, err error) {
```
Comment (nfx) [inline catalog/resource_storage_credentials.go:36]: `cm` -> `scc`. but why do we need to return it?...

simply make `scc *StorageCredentialConfig` as the argument and return error from post straight away

File: catalog/resource_storage_credentials.go [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+	err = a.client.Post(a.context, "/unity-catalog/storage-credentials", cm, &mi)
+	return
+}
+
+func (a StorageCredentialsAPI) get(id string) (sci StorageCredentialConfig, err error) {
```
Comment (nfx) [inline catalog/resource_storage_credentials.go:41]: `sci` -> `scc`

File: catalog/resource_storage_credentials.go [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+	err = a.client.Get(a.context, "/unity-catalog/storage-credentials/"+id, nil, &sci)
+	return
+}
+
+func (a StorageCredentialsAPI) update(storageCredentialsName string, info StorageCredentialConfig) error {
```
Comment (nfx) [inline catalog/resource_storage_credentials.go:46]: `info` -> `scc`

File: catalog/resource_storage_credentials.go [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+			return nil
+		},
+		Read: func(ctx context.Context, d *schema.ResourceData, c *common.DatabricksClient) error {
+			sci, err := NewStorageCredentialsAPI(ctx, c).get(d.Id())
+			if err != nil {
+				return err
+			}
+			return common.StructToData(sci, s, d)
+		},
+		Update: func(ctx context.Context, d *schema.ResourceData, c *common.DatabricksClient) error {
```
Comment (nfx) [inline catalog/resource_storage_credentials.go:81]: missing test for update happy flow (name + aws role) and force new flow - changing one of the things that could be changeable, but force new.

**Review by nfx — CHANGES_REQUESTED** [review]

File: catalog/resource_storage_credentials.go [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+}
+
+type StorageCredentialConfig struct {
+	// name is actually the id of the storage credentials, it is used for GET and DELETE ops
+	// In a given metastore credentials name is unique
+	Name        string                 `json:"name" tf:"force_new"`
+	Aws         *AwsIamRole            `json:"aws_iam_role,omitempty" tf:"group:access"`
+	Azure       *AzureServicePrincipal `json:"azure_service_principal,omitempty" tf:"group:access"`
+	Comment     string                 `json:"comment,omitempty" tf:"force_new"`
+	Owner       string                 `json:"owner,omitempty" tf:"computed"`
```
[the remark this answers is not in the document]
Comment (stikkireddy) [reply catalog/resource_storage_credentials.go:26]: cannot create storage credentials: CreateStorageCredential credential_info.owner can not be provided. is the error we get when set it during the create request

**Review by nfx — APPROVED** [review]
changes applied

**Pull request merged**